pkg/store/prometheus.go

```diff
@@ -160,15 +160,17 @@ func (p *PrometheusStore) Series(r *storepb.SeriesRequest, s storepb.Store_Serie
 			return err
 		}
 		for _, lbm := range labelMaps {
-			lset := make([]labelpb.ZLabel, 0, len(lbm)+len(extLset))
-			for k, v := range lbm {
-				lset = append(lset, labelpb.ZLabel{Name: k, Value: v})
+			lset := make([]labelpb.ZLabel, 0, len(lbm))
```

Comment:
Instead of trying to optimze map sorting + injection, why not ensuring we don't have unsorted map in the first place?

I think we can change this:

`labelMaps, err := p.client.SeriesInGRPC(s.Context(), p.base, newMatchers, r.MinTime, r.MaxTime)`

To return us a sorted list of labels. And we can do that because this call is making a HTTP request to /api/v1/series on Prometheus which by design, it returns sorted series (!).

the actual work to do is in json parser. That JSON Map sorted, but Golang parse that as unsorted map - so we need enhance parser (Unmarhsaller) to parse it as sorted labels.Labels (: